service/resource/endpoint/current.go

```diff
@@ -0,0 +1,38 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/canceledcontext"
+)
+
+const (
+	ipAnnotation      = "endpoint.kvm.giantswarm.io/ip"
+	serviceAnnotation = "endpoint.kvm.giantswarm.io/service"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+
+	pod, err := toPod(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	annotations, err := getAnnotations(*pod, []string{ipAnnotation, serviceAnnotation})
+	if err != nil {
```

Comment:
We should never blindly act on arbitrary errors. The error of interest here is a very specific one. In case we receive it we act accordingly. In case we receive an unexpected error we should return it to the stack's caller.

Reply:
I handled the errors properly now, totally forgot that other errors could happen.